okhttp-tls/src/main/kotlin/okhttp3/tls/internal/der/DerReader.kt

```diff
@@ -129,11 +129,16 @@ internal class DerReader(source: Source) {
       (length0 and 0b1000_0000) == 0b1000_0000 -> {
         // Length specified over multiple bytes.
         val lengthBytes = length0 and 0b0111_1111
+        if (lengthBytes > 8) throw IOException("Length encoded with more than 8 bytes is not supported")
+
         var lengthBits = source.readByte().toLong() and 0xff
```

Comment:
Missing check: This byte must never be 0. It must have the highest bit set if lengthBytes == 1.